### ISSUE TITLE
Debugger does not support breakpoint Expression that compares a Guid, but the error is buried in the Locals section...

### ISSUE THREAD

**Issue by dnpb** [microsoft/AL#6840]
**1. Describe the bug**
I just wasted an hour or two of my life, trying to determine why conditional breakpoints seemingly didn't work, when in fact it appears to be a limitation of the AL extension for VSC - but one that's very quietly reported, nowhere useful, the developer might never see it!!

**2. To Reproduce**
1. Have some code that you need to debug, but you need to only break when a condition is met involving a GUID, e.g. in a loop.
2. Create a breakpoint in VSC and add a condition Expression to it, e.g. `IsNullGuid(SomeRecord.SomeField)`
3. Run the debugger and see that the breakpoint is always hit, even when `SomeRecord.SomeField` is clearly **not** a `NullGuid`.

**3. Expected behavior**
* I expected the Expression to work.
* I now expect **at least** an error to be reported somewhere I can see it / am not likely to miss it... i.e. in a popup message box, etc.

**4. Actual behavior**
* There is an error, but it's buried at the bottom of the Locals tab as an `_ALException`:
![image](https://user-images.githubusercontent.com/63455896/142446186-87a15814-1a24-4090-b5b4-947bef4f1179.png)
![image](https://user-images.githubusercontent.com/63455896/142446195-32735189-abc9-47f2-9806-2a78b8452115.png)
* This exception should be caught by VSC and reported clearly and unambiguously to the user.
* But instead, the debugger continues running and stopping on the breakpoint, despite the condition not being met / being invalid.

**5. Versions:**
- AL Language:
```none
Name: AL Language
Id: ms-dynamics-smb.al
Description: AL development tools for Dynamics 365 Business Central
Version: 8.2.545335
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-dynamics-smb.al
```
- Visual Studio Code: 1.62.2
- Business Central: 
- List of Visual Studio Code extensions that you have installed:
![image](https://user-images.githubusercontent.com/63455896/142445535-d14fc1ba-94d7-4d7f-91eb-0117347a7edf.png)

**Comment by kalberes** [comment]
There is limited conditional breakpoint support today in the debugger. we support max as much as the old C/SIDE debugger has supported.
The _ALException is added to the local variables on errors the server sends.

**Comment by dnpb** [comment]
> The _ALException is added to the local variables on errors the server sends.

Yes, as I *eventually* discovered. But this is not at all helpful to users. Until they notice that, they waste hours trying to determine why their expression does not give a visible error but does not work. And why let the continue debugger to run if the language threw an exception? Is it because there's difficulty differentiating AL exceptions from code vs. from the platform?

**Comment by dnpb** [comment]
I will also say that the debugging experience appears to be very poorly documented or at least poorly searchable in this aspect. I found no relevant results for searches like _Business Central debugger breakpoint condition OR expression_. Something like this should be clearly documented as a gotcha until it's fixed, and be optimised to actually appear in search engines.

**Comment by dnpb** [comment]
Also, am I correct that Hit Count and Log Message type breakpoints are not supported at all by AL? If not, then please can they be disabled in the VSC GUI?

**Comment by kalberes** [comment]
> And why let the debugger continue to run if the language threw an exception? Is it because there's difficulty differentiating AL exceptions from code vs. from the platform?

It was not the language throwing an exception. But the debug engine. Not all exceptions stop debugging.
On the documentation issue.  Agree that it would be nice that we document something you expect from other debug engines to work and doe snot work here. Though there is nothing new here. I can see it the other way around had we supported complex expressions in conditional breakpoint evaluation that should have been documented. Else we can just have a huge list. Why don't you also ask why don't we support intellisense when adding a watch or using the debug console.  Also what about hit counts, log messages. Why don't we document that we do not support that? The list is big.

**Comment by jhoek** [comment]
Spent some time today trying to figure out why "Log Message" didn't appear to produce any output when I saw that the breakpoint's red dot in the editor gutter has a tooltip that tells users that breakpoints of this particular type are not supported. 

I feel that "Log Message" would greatly benefit my debugging experience. Thank you in advance for (re)considering adding this to the VS Code extension!
